### PR TITLE
added more text to define and examplify the type claim

### PR DESCRIPTION
**Pull request by tlodderstedt**

### CODE REVIEW

**Comment by mprorock** [comment]
I worry about collisions and confusion with `typ` (and with `cty`) - those of us that work with this stuff will likely understand the difference, but there is probably a better descriptor we could use than `type` for this reason.  

Maybe specifying "schema" or "schemaName" or "sn" (for nice short form) would be more useful here

File: draft-terbu-oauth-sd-jwt-vc.md [hunk 1]
```diff
@@ -197,18 +197,20 @@ SD-JWT VCs.
 
 #### `type` claim {#type-claim}
 
-This specification defines the JWT claim `type`. The `type` claim is used
-to express the type of the JSON object that is secured by the
-JWT. The `type` value MUST be a case-sensitive `StringOrURI` value.
+This specification defines the JWT claim `type`. The `type` claim value is the
+identifier for a pre-defined bundle of claims associated with certain
+policies for enrollment and issuance that the respective SD-JWT VC
+conforms to. The `type` value MUST be a case-sensitive `StringOrURI` value.
```
Comment (Sakurann) [inline draft-terbu-oauth-sd-jwt-vc.md:203]:
```suggestion
This specification defines the JWT claim `type`. The `type` claim value is the
identifier for a pre-defined bundle of claims that the respective SD-JWT VC
conforms to. The `type` value MUST be a case-sensitive `StringOrURI` value.
```

I don't really understand what `associated with certain policies for enrollment and issuance` means..?

File: draft-terbu-oauth-sd-jwt-vc.md [hunk 2]
```diff
@@ -197,18 +197,20 @@ SD-JWT VCs.
...
 The following is a non-normative example of how `type` is used to express
 a type:
 
 ```
 {
-  "type": "SomeType"
+  "type": "IdentityCredential"
 }
 ```
+The example type `IdentityCredential` is defined as containing at least the claims `given_name`, `family_name`, `birthdate`, and `address` as defined in the in the "JSON Web Token Claims" registry as defined in [@!RFC7519]. SD-JWT VCs of this type might also contain the claims `email` and `phone_number` defined in the registry. Additionally, the type is defined to may include claims `is_over_18`, `is_over_21`, and `is_over_65`.
```
Comment (Sakurann) [inline draft-terbu-oauth-sd-jwt-vc.md:213]:
```suggestion
SD-JWT VC of a type `IdentityCredential` is defined as containing 
- mandatory JWT claims `given_name`, `family_name`, `birthdate`, and `address` registered in the "JSON Web Token Claims" registry created by [@!RFC7519],
- optional JWT claims `email` and `phone_number` defined in the same registry, and
- optional claims `is_over_18`, `is_over_21`, and `is_over_65` defined by this Credential type.
```

I think this is what this text wants to say...?

**Review by Sakurann — DISMISSED** [review]
I agree with the direction would just like to see a bit more clarity..

File: draft-terbu-oauth-sd-jwt-vc.md (hunk 1 above)
Comment (awoie) [reply draft-terbu-oauth-sd-jwt-vc.md:203]: I agree, I don't fully understand either I think I have a clue but imo, it sounds a bit too difficult to parse for most readers.

Comment (awoie) [inline draft-terbu-oauth-sd-jwt-vc.md:203]:
```suggestion
This specification defines the JWT claim `type`. The `type` value
specifies the type of the VC represented by the SD-JWT VC. The
type determines the presence of JWT claims in the 
unsecured payload of the SD-JWT VC. The type also indicates which
JWT claims can have corresponding entries in the `_sd` claim and
are selective disclosable. The `type` value MUST be a case-sensitive
`StringOrURI` value.
```

File: draft-terbu-oauth-sd-jwt-vc.md (hunk 2 above)
Comment (awoie) [inline draft-terbu-oauth-sd-jwt-vc.md:213]:
```suggestion
For example, a type `IdentityCredential` can be defined that an SD-JWT VC of type `IdentityCredential` requires at least the registered JWT claims `given_name`, `family_name`, `birthdate`, and `address` in the unsecured payload. Additionally, the unsecured payload might optionally contain the registered JWT claims `email` and `phone_number`, and the private claims `is_over_18`, `is_over_21`, and `is_over_65`. The type might also indicate that `given_name`, `family_name`, `birthdate`, `address` can have corresponding entries in the `_sd` claim and are selective disclosable. 
```

**Review by awoie — CHANGES_REQUESTED** [review]
@tlodderstedt made some suggestions

File: draft-terbu-oauth-sd-jwt-vc.md (hunk 1 above)
Comment (awoie) [reply draft-terbu-oauth-sd-jwt-vc.md:203]: I'd prefer my suggestion below.

Comment (danielfett) [reply draft-terbu-oauth-sd-jwt-vc.md:203]: Rewording based on @awoie's proposal and incorporating Torsten's ideas.

```suggestion
This specification defines the JWT claim `type`. The `type` value
MUST be a case-sensitive `StringOrURI` value serving as an identifier
for the type of the SD-JWT VC. A type defines which claims MAY or MUST appear in the 
Unsecured Payload of the SD-JWT VC and whether they are
selectively disclosable or not. This specification does not define any `type` values; instead
it is expected that ecosystems using SD-JWT VCs define such values including
the semantics of the respective claims and associated rules (e.g., policies for issuing and
accepting credentials of a certain type).
```

File: draft-terbu-oauth-sd-jwt-vc.md (hunk 2 above)
Comment (danielfett) [reply draft-terbu-oauth-sd-jwt-vc.md:213]:
```suggestion
For example, a type `IdentityCredential` can be defined such that at least the registered JWT claims `given_name`, `family_name`, `birthdate`, and `address` must appear in the Unsecured Payload. Additionally, the registered JWT claims `email` and `phone_number`, and the private claims `is_over_18`, `is_over_21`, and `is_over_65` may be used. The type might also indicate that `given_name`, `family_name`, `birthdate`, `address` can be selectively disclosable.
```

**Review by danielfett — CHANGES_REQUESTED** [review]
Proposed new wordings.

**Review by awoie — APPROVED** [review]
lgtm after we applied @danielfett suggestion

File: draft-terbu-oauth-sd-jwt-vc.md [hunk 3]
```diff
@@ -197,18 +197,24 @@ SD-JWT VCs.
 
 #### `type` claim {#type-claim}
 
-This specification defines the JWT claim `type`. The `type` claim is used
-to express the type of the JSON object that is secured by the
-JWT. The `type` value MUST be a case-sensitive `StringOrURI` value.
+This specification defines the JWT claim `type`. The `type` value
+MUST be a case-sensitive `StringOrURI` value serving as an identifier
+for the type of the SD-JWT VC. A type defines which claims may or must appear in the
```
Comment (danielfett) [inline draft-terbu-oauth-sd-jwt-vc.md:202]:
```suggestion
for the type of the SD-JWT VC. A type defines which claims may or must appear in the
```

**Pull request merged**